### DISCUSSION TITLE
How to use the highlighter crate with the typescript crate

### DISCUSSION THREAD

**Discussion by LaylBongers** [tree-sitter/tree-sitter#1331] [Q&A] [answered]
I can't figure out how to use the highlighter API with the typescript crate to produce correct highlighting:

![the-machinery_3IJfwzg4lK](https://user-images.githubusercontent.com/3094382/128763925-51e8a6ce-3de0-4b0e-b9ca-cf5c3c3b0205.png)

```rust
let config = HighlightConfiguration::new(
    tree_sitter_typescript::language_typescript(),
    tree_sitter_typescript::HIGHLIGHT_QUERY,
    "",
    tree_sitter_typescript::LOCALS_QUERY,
).unwrap();
```

I posted this issue over here: https://github.com/tree-sitter/tree-sitter-typescript/issues/176
But this was considered not an issue with the typescript crate specifically.

Am I using the API wrong? Are the names different? It seems to be working fine with the JavaScript grammar crate.

**Comment by maxbrunsfeld** [comment]
The typescript highlight query is intended to be used as an extension of the JavaScript highlight query. You can combine it with the JavaScript query just by a string concatenation. To see the order in which they are concatenated, check the “highlights”property in the package.json of tree-sitter-typescript.

**Reply by LaylBongers** [reply]
I have tried concatenating the highlight and locals together:
```rust
let mut highlight = tree_sitter_typescript::HIGHLIGHT_QUERY.to_string();
highlight.push_str(tree_sitter_javascript::HIGHLIGHT_QUERY);
let mut locals = tree_sitter_typescript::LOCALS_QUERY.to_string();
locals.push_str(tree_sitter_javascript::LOCALS_QUERY);
```
However this unfortunately produces an error:
```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: QueryError { row: 20, column: 1, offset: 524, message: "required_parameter", kind: NodeType }'
```

**Reply by LaylBongers** [reply]
I doublechecked my code and I found that I was using the wrong `Language` still, but updating that I get a similar error regardless:
```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: QueryError { row: 142, column: 2, offset: 2842, message: "  [\n  ^", kind: Structure }',
```

**Comment by LaylBongers** [comment, marked as answer]
This issue has been fixed in a recent update
